build-utils/git-version-stamp.sh

```diff
@@ -17,7 +17,8 @@ if [ -z "$STAMP" ]; then
     die "Missing STAMP: $1"
 fi
 
-CURRENT=`git describe --dirty 2>/dev/null`
+CURRENT=`git describe 2>/dev/null | awk -F- '{print(($2=="")?($1".0"):($1"."$2))}'`
```

Comment:
Since `awk` is, well, awk:
"If no `-` appears in the input, append `.0`. Otherwise replace the first `-` with `.`".

I am not a big fan of this idea. It is purely cosmetic, but previously you could just take the output of `awesome --version` and feed it to `git show` and things to figure out the exact source code version used (well, and `--dirty` informs you about the existence of changes that you would not find this way). After this, finding the right git commit-ish becomes a lot harder.